lib/ansible/modules/monitoring/grafana_dashboard.py

```diff
@@ -0,0 +1,317 @@
+    required: true
+    default: present
+    description:
+      - State of the dasboard.
+      - can be 'present', 'absent', 'export'
+  slug:
+    description:
+      - slug of the dashboard. It's the friendly url name of the dashboard.
+      - When state is present, this parameter can override the slug in the meta section of the json file.
+      - If you want to import a json dashboard exported directly from the interface (not from the api),
+      - you have to specify the slug parameter because there is no meta section in the exported json.
+  path:
+    description:
+      - path to the json file contaning the grafana dashboard to import or export.
+  overwrite:
+    default: false
+    description:
+      - override existing dashboard when state is present.
+  message:
+    description:
+      - Set a commit message for the version history.
+      - Only used when state is present
+  validate_cert:
+    default: false
+    description:
+      - trust or not the grafana api tls certificate.
+'''
+
+EXAMPLES = '''
+---
+- name: import grafana dashboard foo
+  grafana_dashboard:
+    grafana_url: http://grafana.company.com
+    grafana_api_key: XXXXXXXXXXXX
+    state: present
+    message: "updated by ansible"
+    overwrite: true
+    path: /path/to/dashboards/foo.json
+
+- name: export dashboard
+  grafana_dashboard:
+    grafana_url: http://grafana.company.com
+    grafana_api_key: XXXXXXXXXXXX
+    state: export
+    slug: foo
+    path: /path/to/dashboards/foo.json
+'''
+
+RETURN = '''
+---
+slug:
+  description: slug of the created / deleted / exported dashboard.
+  returned: success
+  type: string
+  sample: foo
+'''
+
+import base64
+import json
+import os
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.urls import fetch_url
+
+__metaclass__ = type
+
+
+class GrafanaAPIException(Exception):
+    pass
+
+
+class GrafanaMalformedJson(Exception):
+    pass
+
+
+def grafana_switch_organisation(module, grafana_url, org_id, headers):
+    r, info = fetch_url(module, '{}/api/user/using/{}'.format(grafana_url, org_id), headers=headers, method='POST')
+    if info['status'] != 200:
+        raise GrafanaAPIException('Unable to switch to organization {} : {}'.format(org_id, info))
+
+
+def grafana_dashboard_exists(module, grafana_url, slug, headers):
+    dashboard_exists = False
+    dashboard = {}
+    r, info = fetch_url(module, '{}/api/dashboards/db/{}'.format(grafana_url, slug), headers=headers, method='GET')
+    if info['status'] == 200:
+        dashboard_exists = True
+        dashboard = json.loads(r.read())
+    elif info['status'] == 404:
+        dashboard_exists = False
+    else:
+        raise GrafanaAPIException('Unable to get dashboard {} : {}'.format(slug, info))
+
+    return dashboard_exists, dashboard
+
+
+def grafana_create_dashboard(module, data):
+
+    # define data payload for grafana API
+    with open(data['path'], 'r') as json_file:
+        payload = json.load(json_file)
```

Comment:
Exception handling should be added around here, incase either the file doesn't exist, can't be opened, or cannot be parsed as json.